qiskit/providers/aer/backends/compatibility.py

```diff
@@ -77,19 +107,37 @@ def __eq__(self, other):
             np.allclose(self.data, other.data, rtol=self.rtol, atol=self.atol)
         )
 
+    def __len__(self):
+        _deprecation_warning(self, "density matrices")
+        return self._data.__len__()
+
+    def __iter__(self):
+        _deprecation_warning(self, "density matrices")
+        return self._data.__iter__()
+
+    def __contains__(self, value):
+        _deprecation_warning(self, "density matrices")
+        return self._data.__contains__(value)
+
+    def __reversed__(self):
+        _deprecation_warning(self, "density matrices")
+        return self._data.__reversed__()
+
+    def __getitem__(self, key):
```

Comment:
These should probably be added to Statevector too for vector access `vec[i]`

Reply:
Never mind, i see this is already in statevector (but has some issues there with only working with str or int keys, not slices/ranges/lists)